apps/frontend/src/library/Routes.tsx

```diff
@@ -16,9 +17,22 @@ const Library: React.FC<{}> = () => {
   const { search } = useLocation();
   const history = useHistory();
   const currentUrlParams = new URLSearchParams(search);
+  let filters = currentUrlParams.getAll('filter');
+  const searchQuery = currentUrlParams.get('searchQuery') || '';
+  const [searchQueryDebounce] = useDebounce(searchQuery, 500);
 
   const onChangeSearch = (newQuery: string) => {
-    currentUrlParams.set('query', newQuery);
+    currentUrlParams.set('searchQuery', newQuery);
+    history.replace({ search: currentUrlParams.toString() });
+  };
+  const onChangeFilter = (filter: string) => {
+    currentUrlParams.delete('filter');
```

Comment:
should `filter` be a `Set`?

Reply:
looks like it could make this handling here easier (condition-less)

Reply:
Set doesn't work with arrays 👎

Reply:
No I mean `filters` being a Set

Reply:
convert when retrieving from SearchParams, and convert back when writing to SearchParams